src/adapters/fabric/fabric.js

```diff
@@ -108,12 +108,10 @@ class Fabric extends BlockchainInterface{
             // Retrieve gateway using ccp and options
             const gateway = await e2eUtils.retrieveGateway(fabricSettings.fabric.ccp, opts);
 
-            // Retrieve and return the contract using the network API commands
-            commLogger.info(`Retrieving contract from channel ${context[name].channel} and chaincodeID ${context[name].chaincodeId}`);
-            const network = await gateway.getNetwork(context[name].channel);
-            const contract = await network.getContract(context[name].chaincodeId);
-
-            return {gateway: gateway, contract: contract};
+            // Retrieve and return the network using the network API commands
+            commLogger.info(`Retrieving network from channelName ${context[name]}`);
+            const network = await gateway.getNetwork(context[name]);
+            return {gateway: gateway, network: network, clientIdx};
```

Comment:
Extraction of the specific contract from the Network object requires knowledge of the chaincode name that the client is to use. At this point in the code flow we only have access to information contained in the "context" portion of the network config file, which details the channel:test-tag mapping only. 

There is a requirement for the user wishing to use the network model testing route to perform the contract extraction as part of the `init` phase of the benchmarks. 

This could probably do with some documentation somewhere 😂 I have a fair few tests that I have been running with this network model, so it might be worth me adding these in as a benchmark sample?

Reply:
As you said in the current code flow we only needed to access to information contained in the "context"  portion of the network config file, it would not affect current process. I think I can merge it first then you can start to pack for the npm package and as for the other related network models,  you might add them later :)